src/main/scala/beam/agentsim/agents/PersonAgent.scala

```diff
@@ -634,14 +628,14 @@ class PersonAgent(
     case Event(
         TriggerWithId(AlightVehicleTrigger(tick, vehicleToExit, energyConsumedOption), triggerId),
         data @ BasePersonData(_, _, _ :: restOfCurrentTrip, currentVehicle, _, _, _, _, _, _, _, _)
-        ) if vehicleToExit.equals(currentVehicle.head) =>
+        ) if currentVehicle.headOption.contains(vehicleToExit) =>
       updateFuelConsumed(energyConsumedOption)
       logDebug(s"PersonLeavesVehicle: $vehicleToExit @ $tick")
       eventsManager.processEvent(new PersonLeavesVehicleEvent(tick, id, vehicleToExit))
       holdTickAndTriggerId(tick, triggerId)
       goto(ProcessingNextLegOrStartActivity) using data.copy(
         restOfCurrentTrip = restOfCurrentTrip.dropWhile(leg => leg.beamVehicleId == vehicleToExit),
-        currentVehicle = currentVehicle.tail
+        currentVehicle = currentVehicle.drop(1)
```

Comment:
Observation: its reduce readability a lot and may be error prone.

Reply:
I agree it reduces the readability but is not error prone. drop never throws exception: if iterable is empty it keeps empty